tests/test_chat_handler.py

```
@@ -0,0 +1,457 @@
+def test_cost_guardrail_cap_hit_emits_audit_event(tmp_path: Path):
+    """When a cap fires, the handler emits ``chat_cap_hit`` so the
+    operator can see the rejection in the audit log."""
+    try:
+        check_cost_guardrail(
+            tmp_path,
+            estimated_input_tokens=99_999,
+            profile=_profile(),
+            limits=_limits(input_cap=10_000),
+        )
+    except ChatCapExceeded:
+        pass
```

Ruff (0.15.12)

[warning] Use `contextlib.suppress(ChatCapExceeded)` instead of `try`-`except`-`pass`

Replace `try`-`except`-`pass` with `with contextlib.suppress(ChatCapExceeded): ...`

(SIM105)